x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/parser/promql/PromqlParamsTests.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.esql.parser.promql;
+
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.xpack.esql.parser.EsqlParser;
+import org.elasticsearch.xpack.esql.parser.ParsingException;
+import org.elasticsearch.xpack.esql.plan.logical.promql.PromqlCommand;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.List;
+
+import static org.elasticsearch.xpack.esql.EsqlTestUtils.as;
+import static org.elasticsearch.xpack.esql.EsqlTestUtils.withDefaultLimitWarning;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.nullValue;
+
+public class PromqlParamsTests extends ESTestCase {
+
+    private static final EsqlParser parser = new EsqlParser();
+
+    public void testValidRangeQuery() {
+        PromqlCommand promql = parse("TS test | PROMQL start `2025-10-31T00:00:00Z` end `2025-10-31T01:00:00Z` step 1m (avg(foo))");
+        assertThat(promql.start(), equalTo(Instant.parse("2025-10-31T00:00:00Z")));
+        assertThat(promql.end(), equalTo(Instant.parse("2025-10-31T01:00:00Z")));
+        assertThat(promql.step(), equalTo(Duration.ofMinutes(1)));
+        assertThat(promql.isRangeQuery(), equalTo(true));
+        assertThat(promql.isInstantQuery(), equalTo(false));
+    }
+
+    public void testValidRangeQueryOnlyStep() {
+        PromqlCommand promql = parse("TS test | PROMQL step 1 (avg(foo))");
+        assertThat(promql.start(), nullValue());
+        assertThat(promql.end(), nullValue());
+        assertThat(promql.step(), equalTo(Duration.ofSeconds(1)));
+        assertThat(promql.isRangeQuery(), equalTo(true));
+        assertThat(promql.isInstantQuery(), equalTo(false));
+    }
+
+    public void testValidInstantQuery() {
+        PromqlCommand promql = parse("TS test | PROMQL time `2025-10-31T00:00:00Z` (avg(foo))");
```

Comment:
`foo` is a quoted identifier that is a reference to a field which the parser should translate into an UnresolvedAttribute(foo).
If you want to parse it as a value use define it as a string - "2025..."